packages/legend-application-studio/src/components/editor/editor-group/service-editor/testable/ServiceTestsEditor.tsx

```diff
@@ -143,48 +180,132 @@ const ServiceTestParameterEditor = observer(
           <button
             className={clsx('type-tree__node__type__label', {})}
             tabIndex={-1}
-            title={
-              paramState.varExpression.genericType?.value.rawType.name ?? ''
-            }
+            title={type}
           >
-            {paramState.varExpression.genericType?.value.rawType.name ??
-              'unknown'}
+            {type}
           </button>
         </div>
-        <div className="service-test-editor__setup__parameter__value">
-          <BasicValueSpecificationEditor
-            valueSpecification={paramState.valueSpec}
-            setValueSpecification={(val: ValueSpecification): void => {
-              paramState.updateValueSpecification(val);
-            }}
-            graph={setupState.editorStore.graphManagerState.graph}
-            obseverContext={
-              setupState.editorStore.changeDetectionState.observerContext
-            }
-            typeCheckOption={{
-              expectedType:
-                paramState.varExpression.genericType?.value.rawType ??
-                PrimitiveType.STRING,
-            }}
-            className="query-builder__parameters__value__editor"
-            resetValue={(): void => {
-              paramState.resetValueSpec();
-            }}
-          />
-          <div className="service-test-editor__setup__parameter__value__actions">
-            <button
-              className="btn--icon btn--dark btn--sm"
-              disabled={isReadOnly || paramIsRequired}
-              onClick={(): void => setupState.removeParamValueState(paramState)}
-              tabIndex={-1}
-              title={
-                paramIsRequired ? 'Parameter Required' : 'Remove Parameter'
-              }
-            >
-              <TimesIcon />
-            </button>
-          </div>
-        </div>
+        <>
+          {bindingParamPair ? (
+            <div className="service-test-editor__setup__parameter__code-editor">
+              {isExpanded ? (
+                <div className="service-test-editor__setup__parameter__code-editor__container">
+                  <div className="service-test-editor__setup__parameter__code-editor__container__content">
+                    <CodeEditor
+                      key={paramState.uuid}
+                      inputValue={
+                        (paramState.valueSpec as PrimitiveInstanceValue)
+                          .values[0] as string
+                      }
+                      updateInput={updateParamValue}
+                      isReadOnly={isReadOnly}
+                      language={
+                        bindingParamPair.binding.contentType ===
```

Comment:
dont we have a function for this already ? in binding editor is this what we use ?

Reply:
we should maybe move this logic in a central location getCodeEditorByContentType
can you please do this.